File: py-ratelimiter/golden_repo/internal/utils/validation.go
package utils

import "fmt"

type ValidationError struct {
    Msg string
}

func (e *ValidationError) Error() string {
    return e.Msg
}

func RequireField(m map[string]any, field string) (any, error) {
    v, ok := m[field]
    if !ok {
        return nil, &ValidationError{Msg: fmt.Sprintf("Missing required field: %s", field)}
    }
    return v, nil
}

func RequirePositiveInt(m map[string]any, field string) (int, error) {
    v, err := RequireField(m, field)
    if err != nil {
        return 0, err
    }
    f, ok := v.(float64)
    if !ok || f <= 0 || float64(int(f)) != f {
        return 0, &ValidationError{Msg: fmt.Sprintf("%s must be a positive integer", field)}
    }
    return int(f), nil
}

File: py-ratelimiter/source_repo/src/utils/validation.py
from typing import Any, Dict


class ValidationError(Exception):
    pass


def require_field(data: Dict[str, Any], field: str) -> Any:
    if field not in data:
        raise ValidationError(f"Missing required field: {field}")
    return data[field]


def require_positive_int(data: Dict[str, Any], field: str) -> int:
    value = require_field(data, field)
    if not isinstance(value, int) or value <= 0:
        raise ValidationError(f"{field} must be a positive integer")
    return value
